blog/12-12-2022-advance-hardhat-configration-on-celo-using-plugins/advance-hardhat-configration-on-celo-using-plugins.md

```diff
@@ -0,0 +1,751 @@
+    },
+ 
+    celo: {
+      url: "https://forno.celo.org",
+      accounts: [`${process.env.PRIVATE_KEY}`],
+      chainId: 42220,
+    },
+ 
+  },
+ 
+  solidity: {
+    version: "0.8.9",
+    settings: {          // See the solidity docs for advice about optimization and evmVersion
+      optimizer: {
+        enabled: true,
+        runs: 200
+      },
+      evmVersion: "byzantium"
+     }
+  },
+ 
+};
+```
+To make a more advanced deployment and testing, we will leverage some of the tools we have installed. 
+
+**Modifying "Lock.sol"**
+For this task, we will modify the Lock contract to make it look more advanced with another dependency contract, and use “hardhat-deploy” to deploy the contracts to the “alfajores” network and save the deployment information anywhere we want for later use which is not available to us in the previous deployment method. Secondly, we want the deployment to run only if changes are made to the contracts. In the contract folder, create a new “.sol” file and name it “Beneficiary.sol”.
+Make a new folder as “interface”. Add a new file named “Beneficiary.sol”.
+Rename “Lock.sol” to “Bank.sol”, then paste the following code to their respective destination.
+
+**Bank.sol**
+This is a simple time-lock contract that allows only approved child address to withdraw from the bank. The Bank makes an external gasless call to the “Beneficiaries” contract to verify if the caller is an approved beneficiary. The former depends on the latter.
+
+```bash
+// SPDX-License-Identifier: UNLICENSED
+pragma solidity ^0.8.9;
+ 
+import "./interfaces/IBeneficiary.sol";
+ 
+contract Bank {
+    address public immutable beneficiary;
+ 
+    event Withdrawal(uint amount, address indexed who);
+ 
+    constructor(address _beneficiary) payable {
+        require(
+           _beneficiary != address(0),
+            "Beneficiary is zero address"
+        );
+ 
+        beneficiary = _beneficiary;
+    }
+ 
+    function withdraw() external {
+        require(IBeneficiary(beneficiary).getApproval(msg.sender), "You're not a beneficiary");
+ 
+        emit Withdrawal(address(this).balance, msg.sender);
+ 
+        payable(msg.sender).transfer(address(this).balance);
+    }
+    
+    function getBalance(address who) external view returns(uint256) {
+        return address(who).balance;
+    }
+ 
+}
+
+```
+
+**Beneficiary.sol**
+An independent contract that returns whether the queried address is a beneficiary or not. Owner is able to approve or disapprove the beneficiary. 
+
+```bash
+// SPDX-License-Identifier: UNLICENSED
+pragma solidity ^0.8.9;
+ 
+contract Beneficiaries {
+  address payable immutable owner;
+ 
+  address payable immutable child_1;
+  address payable immutable child_2;
+ 
+  mapping (address => bool) public approval;
+ 
+  constructor (address _child_1, address _child_2) {
+    require(_child_1 != address(0) && _child_2 != address(0), "Children addresses are empty");
+    child_1 = _child_2;
+    child_2 = _child_2;
```

Comment:
As they are payable set in line 367 and 368, this wont work, kindly solve this

Reply:
I made it similar to the file Benificiaries.sol,to check and it worked, but you are better judge, to know what steps to take to make it more better and better